icu4j/main/core/src/main/java/com/ibm/icu/util/IslamicCalendar.java

```diff
@@ -732,6 +780,33 @@ static final double moonAge(long time)
     // Calendar framework
     //----------------------------------------------------------------------
 
+    /**
+     * The handleGetMonthLength function for Islamic Civil calendar.
+     */
+    private int handleGetMonthLengthIslamicCivil(int extendedYear, int month) {
+        return ((month+1) % 2) +
+            ((month == DHU_AL_HIJJAH && civilLeapYear(extendedYear)) ? 30 : 29);
```

Comment:
I find this expression a little hard to read.  I think it might be worth being a little more verbose here, something like:

```
int monthLength = 29;
if (month % 2 == 0) {
    ++monthLength;
}
if (month == DHU_AL_HIJJAH && civilLeapYear(extendedYear)) {
    ++monthLength;
}
return monthLength;
```

Reply:
done